avm/res/relay/namespace/main.bicep

```diff
@@ -0,0 +1,429 @@
+}
+
+module namespace_hybridConnections 'hybrid-connection/main.bicep' = [for (hybridConnection, index) in hybridConnections: {
+  name: '${uniqueString(deployment().name, location)}-HybridConnection-${index}'
+  params: {
+    namespaceName: namespace.name
+    name: hybridConnection.name
+    authorizationRules: contains(hybridConnection, 'authorizationRules') ? hybridConnection.authorizationRules : [
+      {
+        name: 'RootManageSharedAccessKey'
+        rights: [
+          'Listen'
+          'Manage'
+          'Send'
+        ]
+      }
+      {
+        name: 'defaultListener'
+        rights: [
+          'Listen'
+        ]
+      }
+      {
+        name: 'defaultSender'
+        rights: [
+          'Send'
+        ]
+      }
+    ]
+    requiresClientAuthorization: contains(hybridConnection, 'requiresClientAuthorization') ? hybridConnection.requiresClientAuthorization : true
+    userMetadata: hybridConnection.userMetadata
+  }
+}
+]
+
+module namespace_wcfRelays 'wcf-relay/main.bicep' = [for (wcfRelay, index) in wcfRelays: {
+  name: '${uniqueString(deployment().name, location)}-WcfRelay-${index}'
+  params: {
+    namespaceName: namespace.name
+    name: wcfRelay.name
+    authorizationRules: contains(wcfRelay, 'authorizationRules') ? wcfRelay.authorizationRules : [
+      {
+        name: 'RootManageSharedAccessKey'
+        rights: [
+          'Listen'
+          'Manage'
+          'Send'
+        ]
+      }
+      {
+        name: 'defaultListener'
+        rights: [
+          'Listen'
+        ]
+      }
+      {
+        name: 'defaultSender'
+        rights: [
+          'Send'
+        ]
+      }
+    ]
+    relayType: wcfRelay.relayType
+    requiresClientAuthorization: contains(wcfRelay, 'requiresClientAuthorization') ? wcfRelay.requiresClientAuthorization : true
+    requiresTransportSecurity: contains(wcfRelay, 'requiresTransportSecurity') ? wcfRelay.requiresTransportSecurity : true
+    userMetadata: contains(wcfRelay, 'userMetadata') ? wcfRelay.userMetadata : null
+  }
+}
+]
+
+resource namespace_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?? {}) && lock.?kind != 'None') {
+  name: lock.?name ?? 'lock-${name}'
+  properties: {
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot delete or modify the resource or child resources.'
+  }
+  scope: namespace
+}
+
+resource namespace_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [for (diagnosticSetting, index) in (diagnosticSettings ?? []): {
+  name: diagnosticSetting.?name ?? '${name}-diagnosticSettings'
+  properties: {
+    storageAccountId: diagnosticSetting.?storageAccountResourceId
+    workspaceId: diagnosticSetting.?workspaceResourceId
+    eventHubAuthorizationRuleId: diagnosticSetting.?eventHubAuthorizationRuleResourceId
+    eventHubName: diagnosticSetting.?eventHubName
+    metrics: diagnosticSetting.?metricCategories ?? [
```

Comment:
Please update this resource to the latest spec

Reply:
Done: https://github.com/Azure/bicep-registry-modules/pull/1319/commits/9f9c82d23b6aa9322f1837b44c87af20d8b1d4c6